recipes-bsp/imx-mkimage/imx-boot_1.0.bb

```diff
@@ -127,6 +127,9 @@ compile_mx8x() {
 }
 do_compile() {
     # mkimage for i.MX8
+    # mkimage_uboot requires libssl.so.1.1 from ${STAGING_LIBDIR_NATIVE}
+    export LD_LIBRARY_PATH=${STAGING_LIBDIR_NATIVE}:$LD_LIBRARY_PATH
```

Comment:
The best way to fix this kind of issue is providing the wrapper so we don't need to handle it during the binary use in multiple places. You can take a look at an example here:

https://github.com/openembedded/openembedded-core/blob/e9800fecbae1e4565275cd1528397c3cec81bcab/meta/recipes-gnome/gdk-pixbuf/gdk-pixbuf_2.40.0.bb#L110-L112

Reply:
@otavio this is a good suggestion.

I am also working on this issue, I had a same fix with @petegriffin which I pushed a long time ago but was rejected, as: https://github.com/liuming50/meta-freescale/commit/0f957d7db015ad4f2611500e3e85ff8cef6862f3, just reworked it a little bit but did not notice @petegriffin already created one for dunfell until I saw this.

@petegriffin would you like to create a wrapper for mkimage_uboot as suggested by @otavio, this issue is also blocking us.

Reply:
@otavio @petegriffin 

Some investigation:

create_wrapper would not work quite well since it refers to a mkimage.real in current directory but we are copying the link mkimage to ${S} as mkimage_uboot, so there is not a mkimage.real.

but just simply creating a symbolic link instead of copying it to ${S}, could fix the problem, and we should not deploy mkimage_uboot since that's a broken binary, the changes as follows:
```
diff --git a/recipes-bsp/imx-mkimage/imx-boot_1.0.bb b/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
index a54b439d..3e732796 100644
--- a/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
+++ b/recipes-bsp/imx-mkimage/imx-boot_1.0.bb
@@ -98,9 +98,9 @@ compile_mx8m() {
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${UBOOT_DTB_NAME}   ${BOOT_STAGING}
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/u-boot-nodtb.bin-${MACHINE}-${UBOOT_CONFIG} \
                                                              ${BOOT_STAGING}/u-boot-nodtb.bin
-    cp ${STAGING_DIR_NATIVE}/${bindir}/mkimage               ${BOOT_STAGING}/mkimage_uboot
     cp ${DEPLOY_DIR_IMAGE}/${BOOT_TOOLS}/${ATF_MACHINE_NAME} ${BOOT_STAGING}/bl31.bin
     cp ${DEPLOY_DIR_IMAGE}/${UBOOT_NAME}                     ${BOOT_STAGING}/u-boot.bin
+    lnr ${STAGING_DIR_NATIVE}/${bindir}/mkimage              ${BOOT_STAGING}/mkimage_uboot
 }
 compile_mx8() {
     bbnote 8QM boot binary build
@@ -165,7 +165,6 @@ deploy_mx8m() {
     install -m 0644 ${BOOT_STAGING}/signed_hdmi_imx8m.bin    ${DEPLOYDIR}/${BOOT_TOOLS}
     install -m 0755 ${BOOT_STAGING}/${TOOLS_NAME}            ${DEPLOYDIR}/${BOOT_TOOLS}
     install -m 0755 ${BOOT_STAGING}/mkimage_fit_atf.sh       ${DEPLOYDIR}/${BOOT_TOOLS}
-    install -m 0755 ${BOOT_STAGING}/mkimage_uboot            ${DEPLOYDIR}/${BOOT_TOOLS}
 }
 deploy_mx8() {
     install -d ${DEPLOYDIR}/${BOOT_TOOLS}
```

Reply:
@otavio  @liuming50 
I just submitted the pull against master here https://github.com/Freescale/meta-freescale/pull/574. Unfortunately that took a while for me to test, as master builds are broken in other ways that needed to be resolved to get to this breakage :(

@otavio see @liuming50 comments about create_wrapper. Unfortunately I don't really have any more time to spend on this. If using create_wrapper is a blocker can we just merge the optee-test fix?

Reply:
@petegriffin @otavio as I understood, create_wrapper creates wrappers for binaries in ${D} and they being populated to sysroot. If we copy mkimage to ${S}, that would not work? Unless we do a substitution to replace mkimage_uboot with full path to ${STAGING_LIBDIR_NATIVE} in makefile.

Reply:
You can't use full path as it'd fail if loading from `sstate`. You can use the `$STAGING_LIBDIR_NATIVE` environment variable to use the right path.

@petegriffin please split the optee-test fix so it can be merged before.

Reply:
@otavio OK I just updated the master pull request here https://github.com/Freescale/meta-freescale/pull/574  to only contain the optee test commit. Can you also cherry-pick to dunfell?

Sorry I don't think I understand your feedback about using $STAGING_LIBDIR_NATIVE environment variable. Ins't the patch already using this?